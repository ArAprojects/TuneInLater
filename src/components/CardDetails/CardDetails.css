.CardDetails_section {
  background-color: lightslategrey;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.details-container {
  border: 3px solid black;
  width: 70%;
  text-align: center;
}

.details-container div {
  margin-bottom: 10px;
}