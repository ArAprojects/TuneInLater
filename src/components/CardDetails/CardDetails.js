import React from 'react';
import { Link } from 'react-router-dom';
import { Markup } from 'interweave';
import './CardDetails.css';

const CardDetails = ({author_name, artwork_url, book_name, description }) => {
  return (
    <section className="CardDetails_section">
      <Link to={'/'} className='back-btn'>â—€ back</Link>
      <h4>{book_name}</h4>
      <p>{author_name}</p>
      <p><Markup content={description}/></p>
      <img src={artwork_url} />
    </section>
  )
}

export default CardDetails;