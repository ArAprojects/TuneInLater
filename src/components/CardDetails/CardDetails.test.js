import React from 'react';
import { shallow } from 'enzyme';
import CardDetails from './CardDetails';

describe('CardDetails', () => {
  let wrapper;
  const mockAudioBooks = [{
    artwork_url: 'http//:image.png',
    author_name: 'Dr. Seuss',
    book_name: 'The Cat in the Hat',
    description: 'Story Description',
  }];

  beforeEach(() => {
    wrapper = shallow(<CardDetails mockAudioBooks={mockAudioBooks}
    />)
  })

  it('should keep an up-to-date snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })
})