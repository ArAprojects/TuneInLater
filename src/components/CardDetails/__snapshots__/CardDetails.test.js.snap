// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should keep an up-to-date snapshot 1`] = `
<section
  className="CardDetails_section"
>
  <Link
    className="back-btn"
    to="/"
  >
    â—€ back
  </Link>
  <h4 />
  <p />
  <p>
    <Markup
      content=""
      emptyContent={null}
      parsedContent={null}
      tagName="div"
    />
  </p>
  <img />
</section>
`;
